### DISCUSSION TITLE
Route Not Found 404

### DISCUSSION THREAD

**Discussion by msofiullahs** [area17/twill#1313] [Q&A]
Hi guys, I already try to install this CMS and follow all instruction, checked the route list, all twill routes has listed clearly. But, when I try to access it from browser, it gives me 404 page. Anyway, I'm using laravel 8
<img width="1440" alt="Screen Shot 2021-12-07 at 18 39 40" src="https://user-images.githubusercontent.com/69771153/145022789-855693c8-9185-4674-ad70-a8d20660277d.png">
<img width="1440" alt="Screen Shot 2021-12-07 at 18 40 12" src="https://user-images.githubusercontent.com/69771153/145022848-3bfdde05-6913-483b-972e-b98cba42cbdb.png">
<img width="1440" alt="Screen Shot 2021-12-07 at 18 40 23" src="https://user-images.githubusercontent.com/69771153/145022881-3ac7d091-229d-49e1-864c-52c8ee0e6a97.png">

**Comment by kallefrombosnia** [comment]
I think that I already answered you on Discord regarding this issue, but still here is the answer.

Twill needs to be available on a specific endpoint which is declared in the `.env`, so the website URL also needs to be predefined.

Personally, I'm using Homestead for now, if you need some help write to me on Discord.

https://twill.io/docs/#env

**Reply by khrelian** [reply]
Hi @kallefrombosnia, I'm having the same problem. What endpoint do I have to specifically have to declare in the `.env` ? 
I can see all the routes for twill admin panel in route:list, but when accessing them on the browser it returns 404.

Here's my .env vars:

> APP_URL=localhost:8000
> ADMIN_APP_PATH=twilladmin
> ADMIN_APP_URL=localhost:8000

**Reply by kallefrombosnia** [reply]
Then you need to edit the hosts file.

```sh
#/etc/hosts

127.0.0.1 twill.test

```

After running, `php artisan serve` you should be able to access the Twill panel under `twill.test:8000/twilladmin`

```env
# ./.env
ADMIN_APP_PATH=twilladmin
ADMIN_APP_URL=twill.test
```

**Comment by msofiullahs** [comment]
yes, it is, thanks advance

**Comment by ctocph** [comment]
I thought I found the best answer elsewhere...if you get 404 then you can delete the contents of bootstrap/cache (the files only). BUT got some other issues everytime the cache was cleared and could not figure out why.
Turns out Twill uses mcamara/laravel-localization and guess what? You can't use "php artisan optimise", because you have localized routes.

So you need to use 
php artisan route:trans:cache
php artisan route:trans:clear

https://github.com/mcamara/laravel-localization/blob/master/CACHING.md

This took me many hours to find so leaving it here if anyone else fights the same issue.
